AK/Math.h

```diff
@@ -424,48 +432,77 @@ using Trigonometry::tan;
 namespace Exponentials {
 
 template<FloatingPoint T>
-constexpr T log(T x)
+constexpr T log2(T x)
 {
-    CONSTEXPR_STATE(log, x);
+    CONSTEXPR_STATE(log2, x);
 
 #if ARCH(X86_64)
     T ret;
     asm(
-        "fldln2\n"
+        "fld1\n"
         "fxch %%st(1)\n"
         "fyl2x\n"
         : "=t"(ret)
         : "0"(x));
     return ret;
 #else
 #    if defined(AK_OS_SERENITY)
-    // TODO: Add implementation for this function.
-    TODO();
+    if (x == 0)
+        return -Infinity<T>;
+    if (x <= 0 || __builtin_isnan(x))
+        return NaN<T>;
+
+    // FIXME: Make this more accurate, currently this has an absolute maximum error of <0.000456
+    //        If this is done, use it for x86 floats/doubles as well
+    // Approximation using a fitted 5th degree polynomial in its Horner-form
+    // Before replacing this compare to
+    // https://gist.github.com/Hendiadyoin1/f58346d66637deb9156ef360aa158bf9
+    FloatExtractor<T> ext { .d = x };
+    T exponent = ext.exponent - FloatExtractor<T>::exponent_bias;
+
+    // When the mantissa shows 0b00 (implicitly 1.0) we are on a power of 2
+    if (ext.mantissa == 0)
+        return exponent;
+
+    FloatExtractor<T> mantissa_ext {
+        .mantissa = ext.mantissa,
+        .exponent = FloatExtractor<T>::exponent_bias,
+        .sign = ext.sign
+    };
+
+    // (1 <= mantissa < 2)
+    T mantissa = mantissa_ext.d;
+
+    T log2_mantissa = mantissa * (mantissa * (mantissa * (mantissa * (mantissa * static_cast<T>(0.0379969) - static_cast<T>(0.356221)) + static_cast<T>(1.42488)) - static_cast<T>(3.20599)) + static_cast<T>(4.80898)) - static_cast<T>(2.70919);
+
+    return exponent + log2_mantissa;
+
 #    endif
-    return __builtin_log(x);
+    return __builtin_log2(x);
```

Comment:
I think you can remove the `__builtin` calls and call our code instead – the `__builtin` calls were here to not break running lagom programs on arm64 hosts, and now that the TODO is gone we don't necessarily need them any longer.

If you do want to keep them, please put them in an `#else` (in case we ever want to turn on unreachable-code warnings).